File: Algorithms/485. Max Consecutive Ones/kotlin/solution.kt
class Solution {
    fun findMaxConsecutiveOnes(nums: IntArray): Int {
        var result = 0
        var count = 0

        nums.forEach {
            when (it) {
                1 -> {
                    count++
                    result = Math.max(count, result)
                }
                else -> count = 0
            }
        }

        return result
    }
}

File: Algorithms/485. Max Consecutive Ones/php/solution.php
<?php
class Solution
{

    /**
     * @param Integer[] $nums
     * @return Integer
     */
    function findMaxConsecutiveOnes($nums)
    {
        $max = 0;
        $count = 0;
        for ($i = 0; $i < count($nums); $i++) {
            if ($nums[$i] == 1) {
                $count++;
                if ($count > $max) {
                    $max = $count;
                }
            } else {
                $count = 0;
            }
        }
        return $max;
    }
}